features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/OpennmsKafkaProducer.java

```diff
@@ -134,12 +138,31 @@ private void forwardEvent(Event event) {
         if (forwardNodes && event.getNodeid() != null && event.getNodeid() != 0) {
             maybeUpdateNode(event.getNodeid());
         }
-        sendRecord(() -> {
 
-            final OpennmsModelProtos.Event mappedEvent = protobufMapper.toEvent(event).build();
-            LOG.debug("Sending event with UEI: {}", mappedEvent.getUei());
-            return new ProducerRecord<>(eventTopic, mappedEvent.getUei(), mappedEvent.toByteArray());
-        });
+        boolean result = true;
+
+        if (filterEvents) {
+            ExpressionParser parser = new SpelExpressionParser();
+            Expression exp = parser.parseExpression(eventFilter);
+            try {
+                result = exp.getValue(event, Boolean.class);
+            } catch (Exception e) {
+                LOG.error(" Event filter : {} didn't evaluate any result on Event {}", eventFilter,
+                        event.toStringSimple(), e);
+            }
+        }
+        if (result) {
+            sendRecord(() -> {
+
+                final OpennmsModelProtos.Event mappedEvent = protobufMapper.toEvent(event).build();
+                LOG.debug("Sending event with UEI: {}", mappedEvent.getUei());
+                return new ProducerRecord<>(eventTopic, mappedEvent.getUei(), mappedEvent.toByteArray());
+            });
+        } else {
+            LOG.debug("Event {} not forwarded based on filter {} ", event.toStringSimple(), eventFilter);
```

Comment:
Can we wrap this with an `if(LOG.isDebugEnabled())` clause to avoid calling `event.toStringSimple()` unnecessarily?